django/utils/timezone.py

```diff
@@ -295,13 +295,15 @@ def template_localtime(value, use_tz=None):
 
 # Utilities
 
-def localtime(value, timezone=None):
+def localtime(value=None, timezone=None):
```

Comment:
As noted above, `localtime` only works when `USE_TZ` is `True` at this time. A straightforward implementation of `localdate` as `localtime(...).date()` will have the same restriction and thus be consistent.

It may be useful to make these functions fallback reasonably when `USE_TZ` is `False` for the benefit of pluggable applications. In that case, `localtime()` should return:

1. `datetime.datetime.now()` if no `value` is provided
2. `value` if it is provided and it is naive
3. (probably) an exception if `value` is provided and it is aware

The third point is debatable. I tend to be strict in what my code accepts, which may not be a best practice in Python but prevents silent data corruption. It's easy to slip and manipulate the wrong type accidentally when time zones are involved...

No changes will be needed in `localdate`; it'll get a consistent behavior automatically if we make these changes to `localtime`. In any case, that should happen in a separate commit.